packages/web/src/common/store/account/sagas.js

```diff
@@ -192,13 +192,19 @@ export function* fetchAccountAsync({ isSignUp = false }) {
 
 export function* fetchLocalAccountAsync() {
   const localStorage = yield getContext('localStorage')
+  const audiusBackendInstance = yield getContext('audiusBackendInstance')
 
   yield put(accountActions.fetchAccountRequested())
 
+  const audiusLibs = yield call([audiusBackendInstance, audiusBackendInstance.getAudiusLibs])
+  const wallet = yield call([audiusLibs.web3Manager, audiusLibs.web3Manager.getWalletAddress])
   const cachedAccount = yield call([localStorage, 'getAudiusAccount'])
   const cachedAccountUser = yield call([localStorage, 'getAudiusAccountUser'])
   const currentUserExists = yield call([localStorage, 'getCurrentUserExists'])
-  if (cachedAccount && cachedAccountUser && !cachedAccountUser.is_deactivated) {
+
+  const walletMatches = wallet.toLowerCase() === cachedAccountUser.wallet.toLowerCase()
+
+  if (cachedAccount && cachedAccountUser && !cachedAccountUser.is_deactivated && walletMatches) {
```

Comment:
Should we add a cleanup step to remove the values from localStorage?

Reply:
They will get replaced when the new account comes in!